Scripts/check-nested-frameworks.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh -euo pipefail
+
+# Nested frameworks (i.e. having a Frameworks/ folder inside *.app/Frameworks/.framework) is invalid and will make the build be rejected during TestFlight validation.
+# This can happen especially due to an Xcode 12.4 UI bug when linking binary frameworks to the project which always embed the binary (and if you try to change to Do Not Embed it also removes it from linked libraries)
+# This is a bug in Xcode 12.4 UI that is fixed in Xcode 12.5, so fixing nested frameworks to "Do Not Embed" can be fixed using Xcode 12.5, while still continue using Xcode 12.4 for development after the fix.
+
+# This script is intended to be used as a Build Phase on the main app target, as the very last build phase (and especially after the "Embed Frameworks" phase)
+
+NESTED_FMKS_DIRS=$(find "${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}" -name Frameworks -depth 2) 
+if [ -z "$NESTED_FMKS_DIRS" ]; then
+    echo "✅ No nested framework found, you're good to go!"
+else
+    echo "❌ Found nested \`Frameworks\` folder inside frameworks of final bundle."
+    for fmk_dir in $NESTED_FMKS_DIRS; do
+        parent_fmk=$(basename $(dirname $fmk_dir) .framework)
+        nested_fmks=$(cd "${fmk_dir}" && find . -name '*.framework' -depth 1 | sed "s:^./\(.*\)$:\`\1\`:" | tr '\n' ',')
```

Comment:
Super useful comment! Thank you for adding it (and for extracting the script 🙇‍♂️ )

```suggestion
        # `-depth 1` here is so that if `Networking.framework` is embed in `Yosemite` but 
        # `Codegen` is also embed in `Networking`, we don't want to print about the 2nd level 
        # ie the
        # `WooCommerce.app/Frameworks/Yosemite.framework/Frameworks/Networking.framework/Frameworks/Codegen.framework` 
        # – since the nesting of Codegen within Networking will already be printed as a 
        # separate error.
        #
        # The `sed` call is to get rid of the `./` at the beginning of each output line of 
        # `find . -name …`, and also to quote each line between a pair of backticks.
        # 
        # The `tr` call is to join all the lines with a `,`. This will add an extra `,` at the 
        # end of the list, but we'll get rid of this by using bash substitution 
        # `${nested_fmks%,}` when printing the list line 17, to get rid of the suffixed comma.
        nested_fmks=$(cd "${fmk_dir}" && find . -name '*.framework' -depth 1 | sed "s:^./\(.*\)$:\`\1\`:" | tr '\n' ',')
```

Reply:
Thanks! Reworded that comment a bit and pushed those explanations in 93cac05f3